src/metabase/lib/join.cljc

```diff
@@ -483,19 +483,19 @@
     ;; so we break ties by looking at the poisition of the field reference.
     (mbql.u.match/replace condition
       [op op-opts (lhs :guard lib.util/field-clause?) (rhs :guard lib.util/field-clause?)]
-      (let [lhs-aliased (contains? (lib.options/options lhs) :join-alias)
```

Comment:
Would it be better to use `current-join-alias` here so we're not using the `:join-alias` key directly? I think ideally we'd try to have everything use `with-join-alias` and `current-join-alias` instead of pulling those things out by the keys

Reply:
I've missed this comment, sorry. I agree that `current-join-alias` is the way to go.